src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/ClientVersionLimitRule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Checks if <see cref="Block"/> the client is supported by the network.</summary>
+    public class ClientVersionLimitRule : HeaderValidationConsensusRule
```

Comment:
We should probably unit test this one

Reply:
Now that rules are registered at network level I'd suggest starting to move rules close to their network instead of adding more rules to the consensus feature, especially a rule that is so specific to POA.